sdk/core/core/src/az_span.c

```diff
@@ -391,6 +391,45 @@ AZ_NODISCARD az_result az_span_append_int64(az_span * self, int64_t n) {
   return _az_span_builder_append_uint64(self, n);
 }
 
+static AZ_NODISCARD az_result _az_span_builder_append_uint32(az_span * self, uint32_t n) {
+  if (n == 0) {
+    return az_span_append(*self, AZ_SPAN_FROM_STR("0"), self);
```

Comment:
You can call az_span_append_uint8 here and just pass 0

Reply:
You mean az_span_append_byte (which this PR adds)?
Or do you mean that function shall be renamed to az_span_append_uint8 as well?

Look that there might be a catch on the design of this az_span_append_byte function. It is mentioned on the web (not very specifically at all) that some embedded platforms might not support 8 bits, so chars are uint16_t. In that case we would need to review the story here. The easiest for now is assuming that all platforms we will support must have char = uint8_t.

Reply:
Spans are views over byte arrays; not char arrays or strings. So, we always use uint8.